src/math/utils.rs

```diff
@@ -24,14 +24,26 @@ where
     a
 }
 
+/// Resizes the given size to fit into a new size while preserving the aspect ratio.
```

Comment:
I'd probably word this "Returns the dimensions..." Using the word "Resize" makes it sound like the function actually resamples a specific image.

Reply:
Yes, I have been rewriting that sentence a few times... And English is not my mother tongue.
Feel free to suggest a full sentence.

Reply:
How about:
> Returns the dimensions of the largest possible image that would fit inside a canvas of `canvas_size`, while keeping the same aspect ratio from `original_size`. The returned size will be smaller than the canvas size in at most one dimension.

```rust
fn dimensions_to_fit_canvas(original_size: (u32, u32), canvas_size: (u32, u32)) -> (u32, u32) {...}
```

Reply:
Trouble with the use of `canvas` is that the fill mode can return a size larger than the canvas.
The same problem affects @HeroicKatora's suggestion to use `cover`. `cover` will crop enlarged images to fit the canvas.
But the current fill mode will not crop. Might be confusing...

`DynImage::resize_to_fit()` will actually fill then crop, meaning `fill` is still needed.

I also did a bit of searching for prior art.
The best I found so far is https://legacy.imagemagick.org/Usage/resize/#resize. It does not exhibit a usable API but lists all the resize modes possible and provides wording.
I was hoping to find something immediately adoptable but no luck so far.

Anyways, to move on, what about introducing the enum suggested by @HeroicKatora :

```
fn dimensions_to_fit_canvas(original_size: (u32, u32), canvas_size: (u32, u32), resize_mode: ResizeMode) -> (u32, u32) {...}
```

Where `ResizeMode` can be :
* `FIT` (or `CONTAIN`)
* `FILL` (or `COVER`)
* `FILL_OVERFLOW` (or `COVER_NO_CROP`) // current fill mode

Might be a bit over engineered and does not address the `canvas`/`background-size` issues though...